src/components/ComponentDrawer/ComponentDrawer.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useContext } from 'react';
+import { ComponentDrawerItem } from './components/ComponentDrawerItem/ComponentDrawerItem';
+import { NavigationContext } from '../../common/NavigationProvider';
+import { Images } from '../../assets';
+
+export const ComponentDrawer: React.FC = () => {
+  const { visible, setVisibility } = useContext(NavigationContext);
+  return (
+    <div
+      className={` ${visible ? 'visible ' : 'invisible '} 
+       md:visible absolute z-50  grid grid-cols-1 items-center  bg-[#21201E] 
+        justify-start h-[100%] gap-[50px] w-[100%] 
+        md:static md:w-[275px] p-[30px] overflow-y-scroll no-scrollbar  shadow-drawerShadow`}
+    >
+      <div className="flex">
+        <ComponentDrawerItem
+          path="/"
+          logo={Images.Coffee}
+          text={'WATCH'}
+          isLogo={true}
+        />
+      </div>
+      <div className="flex flex-col gap-[60px]">
+        <div className="flex flex-col gap-[30px]">
+          <ComponentDrawerItem logo={Images.Home} text={'Home'} path="/" />
```

Comment:
Please make a data set like this
const data = [
{
  id : 1,
  icon: icon link,
  text: 'custom text',
  path: 'your path'
},
{
  id : 1,
  icon: icon link,
  text: 'custom text',
  path: 'your path'
},
];

use a map function for iteration and that logic should be in another function

Please use loop if you think you can use